### PR TITLE
add tests for link-delete spec according to #3268

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy [hunk 1]
```diff
@@ -662,6 +664,119 @@ class LinkSpec extends HealthCheckSpecification {
...
 
+    @Tidy
+    def "Unable to delete inactive link with flowPath"() {
+        given: "An inactive link with flow on it"
+        def switchPair = topologyHelper.getNeighboringSwitchPair()
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flowHelperV2.addFlow(flow)
+        def flowPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+
+        // "Make all alternative paths unavailable (bring ports down on the source switch)"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy:675]: just make a pinned flow instead?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy [hunk 2]
```diff
@@ -662,6 +664,119 @@ class LinkSpec extends HealthCheckSpecification {
...
+                with(northbound.validateSwitch(swId)) { validation ->
+                    validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                    validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                }
+            }
+        }
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        if (!linkIsDeleted) {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy:768]: `if (linkIsDeleted)`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy:675]: I would like to leave it as is.
it should allow us to notice once system behavior is changed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: thx, fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: good idea, fixed

**Pull request merged**